specs/core/0_beacon-chain.md

```diff
@@ -1758,22 +1821,33 @@ def update_validator_registry(state: BeaconState) -> None:
             update_validator_status(state, index, new_status=EXITED_WITHOUT_PENALTY)
 
 
-    # Calculate the total ETH that has been penalized in the last ~2-3 withdrawal periods
-    period_index = current_slot // COLLECTIVE_PENALTY_CALCULATION_PERIOD
-    total_penalties = (
-        (latest_penalized_exit_balances[period_index]) +
-        (latest_penalized_exit_balances[period_index - 1] if period_index >= 1 else 0) +
-        (latest_penalized_exit_balances[period_index - 2] if period_index >= 2 else 0)
-    )
-
-    # Calculate penalties for slashed validators
-    def to_penalize(index):
-        return state.validator_registry[index].status == EXITED_WITH_PENALTY
-    validators_to_penalize = filter(to_penalize, range(len(validator_registry)))
-    for index in validators_to_penalize:
-        state.validator_balances[index] -= get_effective_balance(state, index) * min(total_penalties * 3, total_balance) // total_balance
+    # Withdraw validators
+    all_indices = range(len(state.validator_registry))
 
-    return validator_registry, latest_penalized_exit_balances, validator_registry_delta_chain_tip
+    def eligible(index):
+        is_exited = state.validator_registry[x].status in {EXITED_WITHOUT_PENALTY, EXITED_WITH_PENALTY}
+        is_greater_than_or_equal_to_min_validator_withdrawal_time = (
+            state.slot >= state.validator_registry[x].latest_status_change_slot >=
+            MIN_VALIDATOR_WITHDRAWAL_TIME
+        )
+        return is_exited and is_greater_than_or_equal_to_min_validator_withdrawal_time
```

Comment:
naming of `is_greater_than_or_equal_to_min_validator_withdrawal_time` seems a bit verbose without adding much to readability. maybe do the following:

```python
is_eligible_for_withdrawal = (
    is_exited and
    state.slot >= state.validator.latest_status_change_slot + MIN_VALIDATOR_WITHDRAWAL_TIME
)
return is_eligible_for_withdrawal
```